modules/bookmark.js

```diff
@@ -0,0 +1,31 @@
+const Augur = require("augurbot"),
+  u = require("../utils/utils");
+
+// Message context menu for bookmarking a message.
+
+const Module = new Augur.Module()
+.addInteractionCommand({name: "Bookmark",
+  commandId: undefined,
+  process: async (interaction) => {
+    try {
+      await interaction?.deferReply({ephemeral: true});
+      let message = await interaction.channel.messages.fetch(interaction.targetId);
+      if (message) {
+        await interaction.reply({content: "I'm sending you a DM!", ephemeral: true});
+        let embed = u.embed()
+          .setAuthor(message.member?.displayName || message.author?.username, message.author?.displayAvatarURL({size: 16}), message.url)
+          .setDescription(message.cleanContent)
+          .setColor(message.member?.displayColor)
+          .setTimestamp(message.createdAt)
+          .setImage(message.attachments.first()?.url);
```

Comment:
If the bookmarked message has more than one attachment, do you want this function to include more than the first?

As well, do you want to include a link to jump to the original?

I'm guessing this is pretty direct from Icarus v4, which would mean you've already faced the 2nd question before. However, the first Q may have wider possibilities than when this was first written.

Reply:
Updated to allow multiple attachments. Link to original is handled via the author link.